hotness/consumers.py

```diff
@@ -796,140 +733,230 @@ def get_dist_tag(self):
+
+                if result_rh != 0:
+                    self._rh_attach_logs(bz, rh_stuff)
+
+                return True
+
+            except Exception as e:
+                self.log.warning("rebase-helper failed with: %s" % str(e))
+                self.log.warning(traceback.format_exc())
+                self._rh_fail(e, bz, rh_stuff)
+                return False
+
+        finally:
+            self.log.debug("Removing %r" % tmp)
+            shutil.rmtree(tmp)
+
+    def _check_rebase(self, old_task_id, new_task_id):
+        """
+        Try to check and compare rawhide and rebased version builds.
+
+        Runs rebase-helper and follows up on the bugzilla accordingly.
+        """
+        old_task = self.old_triggered_task_ids[old_task_id]
+        new_task = self.new_triggered_task_ids[new_task_id]
+
+        bz = old_task[0]
+        upstream = new_task[2]
+
+        if old_task[1] == 'CLOSED':
+            tmp = tempfile.mkdtemp(prefix='thn-rh-', dir='/var/tmp')
+            try:
+                rh_stuff = None
+                try:
+                    result_rh, rh_stuff = self.buildsys.check_rebase(
+                        upstream, old_task_id, new_task_id, tmp)
+                    if new_task[1] == 'CLOSED':
+                        self.bugzilla.follow_up("Scratch builds compared successfully.", bz)
+                    elif new_task[1] == 'FAILED':
+                        self.bugzilla.follow_up("Scratch build of rebased version failed.", bz)
+                    elif new_task[1] == 'CANCELED':
+                        self.bugzilla.follow_up("Scratch build of rebased version "
+                                                "has been canceled.", bz)
+                    self._rh_attach_logs(bz, rh_stuff)
+                except Exception as e:
+                    self.log.warning("rebase-helper failed with: %s" % str(e))
+                    self.log.warning(traceback.format_exc())
+                    self._rh_fail(e, bz, rh_stuff)
+
+            finally:
+                self.log.debug("Removing %r" % tmp)
+                shutil.rmtree(tmp)
+
+        elif old_task[1] == 'FAILED':
+            self.bugzilla.follow_up("Scratch build of rawhide version failed.", bz)
+        elif old_task[1] == 'CANCELED':
+            self.bugzilla.follow_up("Scratch build of rawhide version has been canceled.", bz)
+
+    def _register_tasks(self, package, version, upstream, bz, rh_stuff):
+        """
+        Add tasks initiated by rebase-helper to tracking dicts.
+
+        Task ids are retrieved from rh_stuff dictionary containing data
+        from rebase-helper.
+
+        Returns ids of both tasks.
+        """
+        old_task_id = None
+        new_task_id = None
+
+        for ver in ['old', 'new']:
+            try:
+                task_id = rh_stuff['build_logs']['build_ref'][ver]['koji_task_id']
+                if ver == 'old':
+                    self.old_triggered_task_ids[task_id] = [
```

Comment:
This list would probably be better as a ``namedtuple`` for the sake of clarity. That way, when it's accessed elsewhere you can do something like ``old_task.bugzilla`` or ``new_task.package``. I like to declare any namedtuples I'm making at the top of the module with docblock.